lib/Cro/WebApp/Template.pm6

```diff
@@ -15,20 +15,23 @@ multi render-template(Str $template, $initial-topic --> Str) is export {
 #| Add a path to search for templates. This will be used by both the template and
 #| render-template functions. If the compile-all flag is passed, then all of the
 #| templates will be compiled up front before the function returns. Otherwise, they
-#| will be compiled on first use.
-sub template-location(IO() $location, :$compile-all --> Nil) is export {
+#| will be compiled on first use. If using compile-all, a test parameter can be set
+#| to limit the templates to compile. By default it will exclude entries whose name
+#| has a leading dot (a "hidden" file in a UNIX system). This can be overriden if
+#| necessary by setting it to *.
+sub template-location(IO() $location, :$compile-all, :$test --> Nil) is export {
     my $template-repo = get-template-repository;
     $template-repo.add-location($location);
-    compile-dir($template-repo, $location) if $compile-all;
+    compile-dir($template-repo, $location, :$test) if $compile-all;
 }
 
-sub compile-dir(Cro::WebApp::Template::Repository $template-repo, IO::Path $location --> Nil) {
-    for dir($location) {
+sub compile-dir(Cro::WebApp::Template::Repository $template-repo, IO::Path $location, :$test = { .IO.basename !~~ / ^ '.' / } --> Nil) {
+    for dir($location).grep($test) {
```

Comment:
Defaults are only applied if the parameter is not passed at all, not if it's undefined. So the default clause needs to be on the `:$test` in the exported `template-location` sub instead.

Reply:
Oh, good catch. I've made the change you suggested. Thanks for the comment!

Also: TIL that `.grep(Nil)` makes it return an empty list, which makes perfect sense and means that this code won't die if someone passes in an explicitly undefined value :tada: (although why would they do that...)